add max-width to BigValue titles

yukseltron opened the pull request:
Resolves #494 

- Add a max-width to BigValue comparison titles
- Left align title to end of Value rather than start

Before:
<img width="578" alt="Screen Shot 2022-12-09 at 12 41 16 AM" src="https://user-images.githubusercontent.com/14843458/206632424-19892460-706f-427f-9be1-a7af4686c6a9.png">

After:
<img width="593" alt="Screen Shot 2022-12-09 at 12 40 40 AM" src="https://user-images.githubusercontent.com/14843458/206632827-9318f43d-1f64-4e3c-8fbe-da0e65a1d460.png">

archiewood commented:
Hey, sorry if this is unclear in the issue. 
The issue we're most worried about here is when the text is "too short" not "too long".

This stackblitz hopefully is a better explanation: https://stackblitz.com/edit/evidence-patpt5?file=pages%2Findex.md

yukseltron commented:
Thanks for the stackblitz, that helped clear things up more.

This time, I applied the min-width to the container instead of just the comparison title.

Before:
<img width="561" alt="Screen Shot 2022-12-09 at 10 40 31 AM" src="https://user-images.githubusercontent.com/14843458/206738658-a6dc470f-d270-4fcc-a2c3-d2f24204dd75.png">

After:
<img width="556" alt="Screen Shot 2022-12-09 at 10 42 00 AM" src="https://user-images.githubusercontent.com/14843458/206738933-800afd65-0c3d-42f8-b4e9-4b8e18c24072.png">

However, I wonder if adding a max-width would still be ideal for longer titles/numbers? 

Example of long title with min-width and max-width:

<img width="543" alt="Screen Shot 2022-12-09 at 10 45 09 AM" src="https://user-images.githubusercontent.com/14843458/206739461-7d02c08e-869a-4cac-99b1-21a9acfc58d1.png">

archiewood commented:
I think I dislike the titles spilling across lines as it ruins the horizontal alignment of the BigValue numbers.
I think in that case we should probably just let the thing run. 

From a user perspective they then have a choice:
- Either shorten the title so that the components are evenly spaced
- Or, if you need the longer title, you keep it

We could expose the prop to the user to allow them to override the min-width with a larger value in cases where they want to have generally longer titles, but still keep alignment. 

WDYT?

yukseltron commented:
I like the idea of exposing min-width as a prop. I think it gives more control to the user. I can make the min-width I set as a default for it.

I also think adding a max-width as an optional prop would be nice too. I think it's better to let the user decide how they want to format their titles and worry less about how short the title should be.

archiewood commented:
^ Sounds perfect 👍🏼

yukseltron commented:
Ok, I added style props for both min-width and max-with. The defaults are 25% and none respectively. As well, I updated the documentation for BigValue.

archiewood commented:
Excellent

**I've done:**
- Reduced the min-width default to 18%: allows you to stack 4 in wider viewports, given the default padding.
![CleanShot 2022-12-09 at 13 06 12](https://user-images.githubusercontent.com/58074498/206765455-c167d8c2-aab3-4ecd-8248-a1ac1a838c3e.gif)

**Can you:**
- Add a changeset: https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset

**Two related questions about the props:**
- Defining the min-width and max-width props at ::root - I'm not a css expert here, but does this mean they are global styles that would be accessible to anything in the page?
- I see to edit the min and max width, you have to add them as css style props `--max-width`, rather than js style props `maxWidth` - for the rest of our components, when we expose props we are using js type props, even if they are for styling purposes. For example see the [exported props in the DataTable](https://github.com/evidence-dev/evidence/blob/main/sites/example-project/src/components/viz/DataTable.svelte#L16)

yukseltron commented:
Ok, I've added a changeset and switched it to use JS props. Docs also updated accordingly.

Good catch about the use of `:root`! I incorrectly thought Svelte also scopes it to just that component.

By changing it to JS style props,  the `:root` reference is no longer needed.
